### PR TITLE
[ROCm] Fix an issue with Softmax.

### PR DESCRIPTION
**Pull request by zoranjovanovic-ns**

### CODE REVIEW

File: xla/service/gpu/gpu_compiler.cc [hunk 1]
```diff
@@ -1412,8 +1412,9 @@ absl::Status GpuCompiler::OptimizeHloPostLayoutAssignment(
     // ReductionDimensionGrouper, as that makes matching the softmax pattern
     // harder.
     if (debug_options.xla_gpu_enable_triton_softmax_fusion() &&
-        cuda_cc != nullptr &&
-        cuda_cc->IsAtLeast(se::CudaComputeCapability::AMPERE)) {
+        ((cuda_cc != nullptr &&
+        cuda_cc->IsAtLeast(se::CudaComputeCapability::AMPERE)) ||
+        rocm_cc != nullptr)) {
```
Comment (thomasjoerg) [inline xla/service/gpu/gpu_compiler.cc:1417]: This change enable Triton-based codegen for SoftMax (and other normalization layers) for ROCm. Is that the intention?
Please note that we will `xla_gpu_enable_triton_softmax_fusion` to default to `true` at some point. Will this work for all AMDGPU architectures?

**Review by thomasjoerg — CHANGES_REQUESTED** [review]

File: xla/service/gpu/gpu_compiler.cc (hunk 1 above)
Comment (zoranjovanovic-ns) [reply xla/service/gpu/gpu_compiler.cc:1417]: Yes, it was intention to enable Triton-based codegen for SoftMax (for all AMDGPU architectures).

**Comment by zoranjovanovic-ns** [comment]
Hi @xla-rotation, do I need to do something more to get this merged?

**Comment by golechwierowicz** [comment]
@thomasjoerg can you take another look?

**Review by ezhulenev — APPROVED** [review]

**Comment by thomasjoerg** [comment]
I missed the reply on this PR. I'm really sorry about this!

**Review by thomasjoerg — APPROVED** [review]

**Pull request closed without merging**